homeassistant/components/landisgyr_heat_meter/sensor.py

```diff
@@ -275,10 +295,18 @@ async def async_setup_entry(
         name="Landis+Gyr Heat Meter",
     )
 
-    sensors = []
+    energy_unit = energy_unit_from_data(coordinator.data)
+    if energy_unit == UnitOfEnergy.GIGA_JOULE:
+        exclude_keys = MWH_ONLY_KEYS
+    elif energy_unit == UnitOfEnergy.MEGA_WATT_HOUR:
+        exclude_keys = GJ_ONLY_KEYS
+    else:
+        exclude_keys = set()
```

Comment:
This feels overly complicated.
Maybe it would be better to use `self._attr_entity_registry_enabled_default`.

Something like this could work, and give more control to the end user.
```python
        if (
            description.native_unit_of_measurement
            in {UnitOfEnergy.GIGA_JOULE, UnitOfEnergy.MEGA_WATT_HOUR}
            and self.native_value is None
        ):
            self._attr_entity_registry_enabled_default = False
```

Reply:
Yeah, this is much (much) cleaner. I'm learning a lot here, thanks.